libraries/shared/src/ShapeInfo.cpp

```diff
@@ -49,46 +46,41 @@ void ShapeInfo::setParams(ShapeType type, const glm::vec3& halfExtents, QString
         case SHAPE_TYPE_STATIC_MESH:
             _url = QUrl(url);
             break;
+        case SHAPE_TYPE_BOX:
+        case SHAPE_TYPE_HULL:
+        case SHAPE_TYPE_SIMPLE_HULL:
+            break;
         default:
             break;
     }
-    _doubleHashKey.clear();
 }
 
 void ShapeInfo::setBox(const glm::vec3& halfExtents) {
-    //TODO WL21389:  Should this pointlist clearance added in case
-    //              this is a re-purposed instance?
-    // See https://github.com/highfidelity/hifi/pull/11024#discussion_r128885491
-    _url = "";
+    clear();
     _type = SHAPE_TYPE_BOX;
     setHalfExtents(halfExtents);
-    _doubleHashKey.clear();
 }
 
 void ShapeInfo::setSphere(float radius) {
-    //TODO WL21389: See comment in setBox regarding clearance
-    _url = "";
+    clear();
     _type = SHAPE_TYPE_SPHERE;
     radius = glm::max(radius, MIN_HALF_EXTENT);
     _halfExtents = glm::vec3(radius);
-    _doubleHashKey.clear();
 }
 
 void ShapeInfo::setPointCollection(const ShapeInfo::PointCollection& pointCollection) {
-    //TODO WL21389: May need to skip resetting type here.
+    // DO NOT clear() ShapeInfo here
     _pointCollection = pointCollection;
     _type = (_pointCollection.size() > 0) ? SHAPE_TYPE_COMPOUND : SHAPE_TYPE_NONE;
     _doubleHashKey.clear();
 }
 
 void ShapeInfo::setCapsuleY(float radius, float halfHeight) {
-    //TODO WL21389: See comment in setBox regarding clearance
-    _url = "";
+    clear();
     _type = SHAPE_TYPE_CAPSULE_Y;
     radius = glm::max(radius, MIN_HALF_EXTENT);
     halfHeight = glm::max(halfHeight, 0.0f);
     _halfExtents = glm::vec3(radius, halfHeight, radius);
-    _doubleHashKey.clear();
```

Comment:
The rest of this stuff is cleanup that I noticed while brainstorming on how to provide code that could populate the `ShapeInfo` with data from the `GeometryCache`.

Reply:
I ran into issues when testing where collision wasn't triggering which I narrowed down to this commit.  It took a while to narrow down where the issue was introduced, so I didn't look to in-depthly into it; however, it looks like something's relying on the explicit member specific clear as opposed to the wholesale clearance or something similar.

I didn't include these changes in the updated pr: https://github.com/highfidelity/hifi/pull/11336 due to the fact the issue with collision not firing.